### PR TITLE
Websocket 2.0.0 part1

### PR DESCRIPTION
**Pull request by markt-asf**
Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of wombat-spec-x.y.pdf
- [x] Specification HTML in the form of wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a Release Review, a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification.
The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/websocket/jakarta.websocket-api/2.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-websocket-tck-2.0.0.zip
- [x] The URL of the compatibility certification request issue:
      eclipse-ee4j/websocket-api#361

### CODE REVIEW

**Comment by edbratt** [comment]
LGTM! Thanks for fixing the references.

**Comment by edbratt** [comment]
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [x] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [x] (Optional) Second PR for just apidocs

2. _index.md
  - [x] Link to spec pdf
  - [x] Link to spec html
  - [x] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [x] Name of and link to at least one Compatible Implementation

3. javadocs (See [pull # 235](https://github.com/jakartaee/specifications/pull/235))
  - [x] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [x] Correct spec title
  - [x] Version number of the form x.y, not x.y.z
  - [x] Correct Eclipse copyright line
  - [x] No DRAFT or SNAPSHOT
  - [x] Correct Logo

5. Spec HTML
  - [x] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by keilw** [comment]
@edbratt, @m0mus are you the mentors here? I created  https://github.com/eclipse-ee4j/websocket-api/pull/358 and https://github.com/eclipse-ee4j/websocket-api/pull/359 (for a 2.0.x branch to avoid losing it there) to resolve the module name conflict https://github.com/eclipse-ee4j/websocket-api/issues/357, As I saw, Websocket had already declared an OSGi module extension-point (used e.g. in Eclipse plugins) called "jakarta.websocket" and "jakarta.websocket.server" under the "javax" namespace in Jakarta EE 8, I figured, it seems wise to keep the two declarations in META-INF under the same name than to delete the Jigsaw one.

**Review by edbratt — CHANGES_REQUESTED** [review]
Block inadvertent merging until all checklist items are completed.
Also, pending

- https://github.com/eclipse-ee4j/websocket-api/pull/358
- https://github.com/eclipse-ee4j/websocket-api/pull/355

**Comment by edbratt** [comment]
I recommend changing the TCK link in the summary to the promoted folder copy: https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-websocket-tck-2.0.0.zip

**Comment by markt-asf** [comment]
Updated as requested

File: websocket/2.0/_index.md
Comment (m0mus) [inline websocket/2.0/_index.md:3]: Must be real date. Put something like 2020-10-06 there.

**Comment by m0mus** [comment]
@markt-asf There is also compatibility request is missing. It's the main blocker for now. Everything else looks fine (except tiny issues mentioned above). Are you working on the compatibility request?

**Comment by markt-asf** [comment]
Tomcat was mostly just a placeholder for the CI. We could use Tomcat if necessary (it does pass the TCK) but my understanding is that Tyrus is going to have to file a CCR anyway so my plan is to reference that once available.

**Comment by edbratt** [comment]
Multiple compatible implementations are allowed. It just depends on the implementation projects and if the API team wants to coordinate explicitly. The only requirement is there must be at least one to complete the ballot.

**Comment by m0mus** [comment]
Tyrus compatibility request is created: https://github.com/eclipse-ee4j/websocket-api/issues/361
@markt-asf you can update this PR accordingly and go for the ballot

**Review by edbratt — CHANGES_REQUESTED** [review]
Given the Tyrus CCR, I believe, with the following updates, this PR can be approved. This review is requesting the committer team to please sign-off, or allow me to take these actions:
- [ ] _index.md: Add reference to Tyrus CCR issue https://github.com/eclipse-ee4j/websocket-api/issues/361
- [ ] _index.md: update links to TCK, Sig, SHA, and Pub as follows: 
  - TCK: https://download.eclipse.org/jakartaee/websocket/2.0/jakarta-websocket-tck-2.0.0.zip
  - Sig: https://download.eclipse.org/jakartaee/websocket/2.0/jakarta-websocket-tck-2.0.0.zip.sig
  - SHA: https://download.eclipse.org/jakartaee/websocket/2.0/jakarta-websocket-tck-2.0.0.zip.sha256
  - Pub: https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub

- [ ] Determine the compatible implementations to include. Tyrus RC-1 may be used, according to the CCR referenced above. If the committer team wants to include Tomcat, please provide those details from _index.md (CCR, TCK Summary, Etc.). If you do not wish to use Tomcat, please remove that link, for now (of course, you can wait until Tomcat is ready, totally up to you.) If you approve use of Tyrus 2.0 RC1, please copy paste the download details as provided in the CCR, under the bullet: "Product Name, Version and download URL (if applicable)"

- [ ] Please file a PR to update https://github.com/eclipse-ee4j/jakartaee-api to point at the final API version (currently it's using -RC1).

I think that covers this PR -- almost there!

**Comment by markt-asf** [comment]
OK. Should all be done. Feel free to ping me / this issue if I missed something or got something wrong.

**Review by edbratt — APPROVED** [review]

**Comment by edbratt** [comment]
This all looks good to me. Thank you all! Unless I hear objections, I will start the ballot tomorrow morning (Pacific Time), Oct. 21.

**Comment by markt-asf** [comment]
Fixed missing sign-off as requested off-list. Rebased as well since I was going to have to do a forced push anyway.

**Comment by edbratt** [comment]
> 
> 
> Fixed missing sign-off as requested off-list. Rebased as well since I was going to have to do a forced push anyway.

Thanks Mark. Sorry for the hassle. It all looks good now in this PR!

**Comment by kwsutter** [comment]
One minor comment, but it shouldn't hold up the ballot.  In the Eclipse Release Record, you should add a sentence or two about what this major release of Web Socket 2.0 is bringing to the table.  The EMO is asking for something more than just pointing at the Jakarta EE 9 Release Plan.  Thanks!

**Review by kwsutter — APPROVED** [review]
LGTM.  Looks ready for ballot.  Thanks!

**Comment by kazumura** [comment]
The link of CI in _index.md is CR issue now.  
It should be CI website.

**Comment by edbratt** [comment]
> 
> 
> The link of CI in _index.md is CR issue now.
> It should be CI website.

I considered this and decided all the relevant material was included in the issue. If the committee wishes, we can update the link to the link contained in the issue either now, or when the page is finalized at the conclusion of the ballot.

**Comment by edbratt** [comment]
Voting thread: https://www.eclipse.org/lists/jakarta.ee-spec/msg01182.html

**Comment by edbratt** [comment]
- [x] adds this final checklist to the main PR.
 - [x] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>

**Pull request merged**
